### titanic 
Step 1 ให้นักศึกษาหาโหลด ข้อมูล titanic จาก Github หรือ โหลดมาเข้าเครื่องตัวเอง 
Step 2 นำเข้าข้อมูลสร้าง DataFrame ชื่อ titanic 
Step 3 สำรวจข้อมูลเบื้องต้นพร้อมสรุป โดยเขียนเป็น Comment # 
Step 4 ให้หาข้อมูลเพื่อตอบคำถามต่อไปนี้ 

. การเข้าถึงข้อมูล (Accessing Data with .loc)
แบบฝึกหัด:
    เข้าถึงข้อมูล คอลัมน์ ‘Name’ ทั้งหมด
    เข้าถึงข้อมูล คอลัมน์ ‘Name’ และ ‘Age’ ของผู้โดยสารทุกคน
    แสดงข้อมูล แถวที่ 10 ของ DataFrame (ใช้ loc)
    แสดงข้อมูล แถวที่ 5–10 ของ DataFrame พร้อมคอลัมน์ ‘Name’ และ ‘Survived’
    เข้าถึงข้อมูลของผู้โดยสารที่ มี PassengerId = 50 (ทุกคอลัมน์)
    เข้าถึงข้อมูลของผู้โดยสาร 5 คนแรก เฉพาะคอลัมน์ ‘Name’, ‘Age’, ‘Sex’
    เข้าถึง ชื่อ (Name) และ ค่าโดยสาร (Fare) ของผู้โดยสารแถวที่ 100–110
    เข้าถึงข้อมูล ผู้โดยสารที่รอดชีวิต (Survived = 1) เฉพาะคอลัมน์ ‘Name’ และ ‘Pclass’
    เข้าถึงข้อมูล ผู้โดยสารที่อยู่ชั้น Pclass = 1 เฉพาะคอลัมน์ ‘Name’, ‘Sex’, ‘Age’
    เข้าถึงข้อมูลแบบกำหนด index + column พร้อมกัน (เช่น แถวที่ 15–20 เฉพาะคอลัมน์ ‘Name’, ‘Age’, ‘Survived’)

3. การกรองข้อมูล (Filtering)
แบบฝึกหัด:
    เลือกเฉพาะข้อมูลผู้โดยสารที่ รอดชีวิต (Survived = 1)
    เลือกเฉพาะข้อมูลผู้โดยสารที่ อายุต่ำกว่า 18 ปี
    เลือกเฉพาะข้อมูลผู้โดยสารที่ เป็นผู้หญิง (Sex = ‘female’) และอยู่ชั้น Pclass = 1
    เลือกเฉพาะผู้โดยสารที่ Fare มากกว่า 100
    เลือกเฉพาะผู้โดยสารที่ อายุไม่เกิน 30 ปี และรอดชีวิต


In [1]:
import pandas as pd

In [48]:
df_from_csv = pd.read_csv("./datasets/titanic.csv")
df_from_csv


PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

### ข้อมูลในตารางนี้ เป็นผู้โดยสารของเรือ Titanic ซึ่งมาจาก ชุดข้อมูล Titanic 
เป็นข้อมูลของผู้โดยสารทั้งหมด 891 คน ที่อยู่บนเรือ Titanic รวมทั้งข้อมูลส่วนบุคคลและสถานะการรอดชีวิต เพื่อใช้ในการวิเคราะห์หรือพยากรณ์ว่าใครรอดชีวิตจากอุบัติเหตุเรืออัปปาง

| คอลัมน์         | ความหมาย                                                            |
| --------------- | ------------------------------------------------------------------- |
| **PassengerId** | หมายเลขประจำตัวผู้โดยสาร                                            |
| **Survived**    | สถานะการรอดชีวิต (0 = ไม่รอด, 1 = รอดชีวิต)                         |
| **Pclass**      | ชั้นโดยสาร (1 = ชั้นหนึ่ง, 2 = ชั้นสอง, 3 = ชั้นสาม)                |
| **Name**        | ชื่อผู้โดยสาร                                                       |
| **Sex**         | เพศของผู้โดยสาร (male = ชาย, female = หญิง)                         |
| **Age**         | อายุของผู้โดยสาร                                                    |
| **SibSp**       | จำนวนพี่น้อง/คู่สมรสที่โดยสารมาด้วยกัน                              |
| **Parch**       | จำนวนพ่อแม่/ลูกที่โดยสารมาด้วยกัน                                   |
| **Ticket**      | หมายเลขตั๋ว                                                         |
| **Fare**        | ราคาตั๋ว (ค่าโดยสาร)                                                |
| **Cabin**       | หมายเลขห้องโดยสาร (บางคนไม่มีข้อมูล)                                |
| **Embarked**    | ท่าเรือที่ขึ้นเรือ (S = Southampton, C = Cherbourg, Q = Queenstown) |


In [ ]:
df_from_csv.columns

# เข้าถึงข้อมูล คอลัมน์ ‘Name’ ทั้งหมด
df_from_csv["Name"]

# เข้าถึงข้อมูล คอลัมน์ ‘Name’ และ ‘Age’ ของผู้โดยสารทุกคน
df_from_csv[["Name","Age"]]


Name   Age
0                              Braund, Mr. Owen Harris  22.0
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  38.0
2                               Heikkinen, Miss. Laina  26.0
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  35.0
4                             Allen, Mr. William Henry  35.0
..                                                 ...   ...
886                              Montvila, Rev. Juozas  27.0
887                       Graham, Miss. Margaret Edith  19.0
888           Johnston, Miss. Catherine Helen "Carrie"   NaN
889                              Behr, Mr. Karl Howell  26.0
890                                Dooley, Mr. Patrick  32.0

[891 rows x 2 columns]

In [ ]:
# แสดงข้อมูล แถวที่ 10 ของ DataFrame (ใช้ loc)
df_from_csv.loc[9]

PassengerId                                     10
Survived                                         1
Pclass                                           2
Name           Nasser, Mrs. Nicholas (Adele Achem)
Sex                                         female
Age                                           14.0
SibSp                                            1
Parch                                            0
Ticket                                      237736
Fare                                       30.0708
Cabin                                          NaN
Embarked                                         C
Name: 9, dtype: object

In [ ]:
# แสดงข้อมูล แถวที่ 5–10 ของ DataFrame พร้อมคอลัมน์ ‘Name’ และ ‘Survived’
df_from_csv.loc[4:9, ["Name" , "Survived"]] 

Name  Survived
4                           Allen, Mr. William Henry         0
5                                   Moran, Mr. James         0
6                            McCarthy, Mr. Timothy J         0
7                     Palsson, Master. Gosta Leonard         0
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)         1
9                Nasser, Mrs. Nicholas (Adele Achem)         1

In [ ]:
# เข้าถึงข้อมูลของผู้โดยสารที่ มี PassengerId = 50 (ทุกคอลัมน์)
Titanic = df_from_csv
Titanic_PassengerId_50 = df_from_csv[df_from_csv["PassengerId"]==50]
Titanic_PassengerId_50


PassengerId  Survived  Pclass  \
49           50         0       3   

                                             Name     Sex   Age  SibSp  Parch  \
49  Arnold-Franchi, Mrs. Josef (Josefine Franchi)  female  18.0      1      0   

    Ticket  Fare Cabin Embarked  
49  349237  17.8   NaN        S

In [40]:
# เข้าถึงข้อมูลของผู้โดยสาร 5 คนแรก เฉพาะคอลัมน์ ‘Name’, ‘Age’, ‘Sex’
Titanic = df_from_csv
Titanic.loc[0:4,["Name","Age","Sex"]]

Name   Age     Sex
0                            Braund, Mr. Owen Harris  22.0    male
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  38.0  female
2                             Heikkinen, Miss. Laina  26.0  female
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  35.0  female
4                           Allen, Mr. William Henry  35.0    male

In [52]:
# เข้าถึง ชื่อ (Name) และ ค่าโดยสาร (Fare) ของผู้โดยสารแถวที่ 100–110
Titanic.loc[99:109, ["Name", "Fare"]]


Name     Fare
99                  Kantor, Mr. Sinai  26.0000
100           Petranec, Miss. Matilda   7.8958
101  Petroff, Mr. Pastcho ("Pentcho")   7.8958
102         White, Mr. Richard Frasar  77.2875
103        Johansson, Mr. Gustaf Joel   8.6542
104    Gustafsson, Mr. Anders Vilhelm   7.9250
105             Mionoff, Mr. Stoytcho   7.8958
106  Salkjelsvik, Miss. Anna Kristine   7.6500
107            Moss, Mr. Albert Johan   7.7750
108                   Rekic, Mr. Tido   7.8958
109               Moran, Miss. Bertha  24.1500

In [53]:
#  เข้าถึงข้อมูล ผู้โดยสารที่รอดชีวิต (Survived = 1) เฉพาะคอลัมน์ ‘Name’ และ ‘Pclass’
Titanic[Titanic['Survived'] == 1][['Name', 'Pclass']]


Name  Pclass
1    Cumings, Mrs. John Bradley (Florence Briggs Th...       1
2                               Heikkinen, Miss. Laina       3
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)       1
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)       3
9                  Nasser, Mrs. Nicholas (Adele Achem)       2
..                                                 ...     ...
875                   Najib, Miss. Adele Kiamie "Jane"       3
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)       1
880       Shelley, Mrs. William (Imanita Parrish Hall)       2
887                       Graham, Miss. Margaret Edith       1
889                              Behr, Mr. Karl Howell       1

[342 rows x 2 columns]

In [55]:
# เข้าถึงข้อมูล ผู้โดยสารที่อยู่ชั้น Pclass = 1 เฉพาะคอลัมน์ ‘Name’, ‘Sex’, ‘Age’
Titanic[Titanic["Pclass"]== 1] [["Name","Sex","Age"]]

Name     Sex   Age
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0
6                              McCarthy, Mr. Timothy J    male  54.0
11                            Bonnell, Miss. Elizabeth  female  58.0
23                        Sloper, Mr. William Thompson    male  28.0
..                                                 ...     ...   ...
871   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)  female  47.0
872                           Carlsson, Mr. Frans Olof    male  33.0
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0
887                       Graham, Miss. Margaret Edith  female  19.0
889                              Behr, Mr. Karl Howell    male  26.0

[216 rows x 3 columns]

In [ ]:
# เข้าถึงข้อมูลแบบกำหนด index + column พร้อมกัน (เช่น แถวที่ 15–20 เฉพาะคอลัมน์ ‘Name’, ‘Age’, ‘Survived’)


## 3. การกรองข้อมูล (Filtering)
แบบฝึกหัด

In [63]:
# เลือกเฉพาะข้อมูลผู้โดยสารที่ รอดชีวิต (Survived = 1)
Titanic[Titanic["Survived"] == 1]

PassengerId  Survived  Pclass  \
1              2         1       1   
2              3         1       3   
3              4         1       1   
8              9         1       3   
9             10         1       2   
..           ...       ...     ...   
875          876         1       3   
879          880         1       1   
880          881         1       2   
887          888         1       1   
889          890         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
..                                                 ...     ...   ...    ...   
875                   Najib, Miss. Adele Kiamie "Jane"  female  15.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
880       Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch            Ticket     Fare Cabin Embarked  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
8        2            347742  11.1333   NaN        S  
9        0            237736  30.0708   NaN        C  
..     ...               ...      ...   ...      ...  
875      0              2667   7.2250   NaN        C  
879      1             11767  83.1583   C50        C  
880      1            230433  26.0000   NaN        S  
887      0            112053  30.0000   B42        S  
889      0            111369  30.0000  C148        C  

[342 rows x 12 columns]

In [57]:
# เลือกเฉพาะข้อมูลผู้โดยสารที่ อายุต่ำกว่า 18 ปี
Titanic_Age_more_than_18 = Titanic["Age"]<18
Titanic_Age_more_than_18




0      False
1      False
2      False
3      False
4      False
       ...  
886    False
887    False
888    False
889    False
890    False
Name: Age, Length: 891, dtype: bool

In [71]:
# เลือกเฉพาะข้อมูลผู้โดยสารที่ เป็นผู้หญิง (Sex = ‘female’) และอยู่ชั้น Pclass = 1
Titanic[(Titanic["Sex"]=="female") & (Titanic["Pclass"] == 1)]

# f ต้องเป็นตัวพิมพ์เล็ก


PassengerId  Survived  Pclass  \
1              2         1       1   
3              4         1       1   
11            12         1       1   
31            32         1       1   
52            53         1       1   
..           ...       ...     ...   
856          857         1       1   
862          863         1       1   
871          872         1       1   
879          880         1       1   
887          888         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
11                            Bonnell, Miss. Elizabeth  female  58.0      0   
31      Spencer, Mrs. William Augustus (Marie Eugenie)  female   NaN      1   
52            Harper, Mrs. Henry Sleeper (Myna Haxtun)  female  49.0      1   
..                                                 ...     ...   ...    ...   
856         Wick, Mrs. George Dennick (Mary Hitchcock)  female  45.0      1   
862  Swift, Mrs. Frederick Joel (Margaret Welles Ba...  female  48.0      0   
871   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)  female  47.0      1   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   

     Parch    Ticket      Fare Cabin Embarked  
1        0  PC 17599   71.2833   C85        C  
3        0    113803   53.1000  C123        S  
11       0    113783   26.5500  C103        S  
31       0  PC 17569  146.5208   B78        C  
52       0  PC 17572   76.7292   D33        C  
..     ...       ...       ...   ...      ...  
856      1     36928  164.8667   NaN        S  
862      0     17466   25.9292   D17        S  
871      1     11751   52.5542   D35        S  
879      1     11767   83.1583   C50        C  
887      0    112053   30.0000   B42        S  

[91 rows x 12 columns]

In [76]:
# เลือกเฉพาะผู้โดยสารที่ Fare มากกว่า 100
Titanic_fare_more_than_100 = Titanic["Fare"]>100
Titanic_fare_more_than_100

# แล้วทำไมอันนี้ F เป็นตัวพิมพ์ใหญ่ ?????????????????++++++


1      False
2      False
3      False
8      False
9      False
       ...  
875    False
879    False
880    False
887    False
889    False
Name: Fare, Length: 342, dtype: bool

In [81]:
# เลือกเฉพาะผู้โดยสารที่ อายุไม่เกิน 30 ปี และรอดชีวิต
Titanic[(Titanic["Age"] <= 30) & (Titanic["Survived"] == 1)]

# ยากมากกกกกกกกกกกกกกกกกก

PassengerId  Survived  Pclass  \
2              3         1       3   
8              9         1       3   
9             10         1       2   
10            11         1       3   
22            23         1       3   
..           ...       ...     ...   
874          875         1       2   
875          876         1       3   
880          881         1       2   
887          888         1       1   
889          890         1       1   

                                                  Name     Sex   Age  SibSp  \
2                               Heikkinen, Miss. Laina  female  26.0      0   
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
10                     Sandstrom, Miss. Marguerite Rut  female   4.0      1   
22                         McGowan, Miss. Anna "Annie"  female  15.0      0   
..                                                 ...     ...   ...    ...   
874              Abelson, Mrs. Samuel (Hannah Wizosky)  female  28.0      1   
875                   Najib, Miss. Adele Kiamie "Jane"  female  15.0      0   
880       Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch            Ticket     Fare Cabin Embarked  
2        0  STON/O2. 3101282   7.9250   NaN        S  
8        2            347742  11.1333   NaN        S  
9        0            237736  30.0708   NaN        C  
10       1           PP 9549  16.7000    G6        S  
22       0            330923   8.0292   NaN        Q  
..     ...               ...      ...   ...      ...  
874      0         P/PP 3381  24.0000   NaN        C  
875      0              2667   7.2250   NaN        C  
880      1            230433  26.0000   NaN        S  
887      0            112053  30.0000   B42        S  
889      0            111369  30.0000  C148        C  

[166 rows x 12 columns]